db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/Predicate.java

```diff
@@ -81,6 +81,8 @@
     // getPredScopedForResultSet() method of this class for more.
     private boolean scoped;
 
+    private boolean matchIndexExpression;
```

Comment:
Can you reset the flag matchIndexExpression in clearScanFlags()?

Reply:
Done.